pkg/vpn_client/iptables.go

```diff
@@ -35,6 +37,52 @@ func SetIPTableRules(log logr.Logger, cfg config.VPNClient) error {
 				if err != nil {
 					return err
 				}
+				if !cfg.IsHA {
+					log.Info("setting up double NAT IPv4 iptables rules")
+					ipv4PodNetworks := network.GetByIPFamily(cfg.ShootPodNetworks, network.IPv4Family)
+					if len(ipv4PodNetworks) > 1 {
+						return fmt.Errorf("exactly one v4 pod network is supported. v4 pod networks: %s", ipv4PodNetworks)
+					}
+					ipv4ServiceNetworks := network.GetByIPFamily(cfg.ShootServiceNetworks, network.IPv4Family)
+					if len(ipv4ServiceNetworks) > 1 {
+						return fmt.Errorf("exactly one v4 service network is supported. v4 service networks: %s", ipv4ServiceNetworks)
+					}
+					ipv4NodeNetworks := network.GetByIPFamily(cfg.ShootNodeNetworks, network.IPv4Family)
+					if len(ipv4NodeNetworks) > 1 {
+						return fmt.Errorf("exactly one v4 node network is supported. v4 node networks: %s", ipv4NodeNetworks)
+					}
+
+					for _, nw := range ipv4PodNetworks {
+						err = ipTable.AppendUnique("nat", "PREROUTING", "--in-interface", forwardDevice, "-d", constants.ShootPodNetworkMapped, "-j", "NETMAP", "--to", nw.String())
+						if err != nil {
+							return err
+						}
+						err = ipTable.AppendUnique("nat", "POSTROUTING", "--out-interface", forwardDevice, "-s", nw.String(), "-j", "NETMAP", "--to", constants.ShootPodNetworkMapped)
+						if err != nil {
+							return err
+						}
+					}
+					for _, nw := range ipv4ServiceNetworks {
+						err = ipTable.AppendUnique("nat", "PREROUTING", "--in-interface", forwardDevice, "-d", constants.ShootServiceNetworkMapped, "-j", "NETMAP", "--to", nw.String())
+						if err != nil {
+							return err
+						}
+						err = ipTable.AppendUnique("nat", "POSTROUTING", "--out-interface", forwardDevice, "-s", nw.String(), "-j", "NETMAP", "--to", constants.ShootServiceNetworkMapped)
+						if err != nil {
+							return err
+						}
+					}
+					for _, nw := range ipv4NodeNetworks {
+						err = ipTable.AppendUnique("nat", "PREROUTING", "--in-interface", forwardDevice, "-d", constants.ShootNodeNetworkMapped, "-j", "NETMAP", "--to", nw.String())
+						if err != nil {
+							return err
+						}
+						err = ipTable.AppendUnique("nat", "POSTROUTING", "--out-interface", forwardDevice, "-s", nw.String(), "-j", "NETMAP", "--to", constants.ShootNodeNetworkMapped)
+						if err != nil {
+							return err
+						}
+					}
```

Comment:
Do we want to unify this into a single loop? Please also add error context.

Reply:
we will keep it this way for easier readability with a little more code.